src/toRedactor.tsx

```diff
@@ -49,6 +49,8 @@ const ELEMENT_TYPES: IJsonToHtmlElementTags = {
     return `<iframe${attrs}></iframe>`
   },
   p: (attrs: any, child: any) => {
+    if(child.includes("<figure"))
+    return `<div${attrs} style="overflow: hidden"><span>${child}</span></div>`
```

Comment:
Why do we need this?

Reply:
So, when a figure is an inline element, we need to wrap it inside a block as it has other text with it. in the case of `p` being the block element, the DOM moves the `figure` tag outside of the `p` tag, giving an unexpected behavior where some text is outside the `p`  tag while some are inside the `p` tag. To prevent this from happening in the client browser, we replace the `p` with the `div`  here.

Reply:
Got it.